securesystemslib/signer.py

```diff
@@ -137,8 +148,154 @@ def to_dict(self) -> Dict:
         }
 
 
+class Key:
```

Comment:
In `in-toto/securesystemslib` we made [`Key`](https://github.com/secure-systems-lab/securesystemslib/compare/master...in-toto:securesystemslib:master#diff-ae2ca657fc834f3033da3502b45f73a8380f6a6cc2fde77fafb3715fa5237c65R12) an abstract interface to support multiple container formats and verification  implementations.

We made your traditional securesystemslib `Key` a subclass called [`SSlibKey`](https://github.com/secure-systems-lab/securesystemslib/compare/master...in-toto:securesystemslib:master#diff-ae2ca657fc834f3033da3502b45f73a8380f6a6cc2fde77fafb3715fa5237c65R46) and added another subclass [`GPGKey`](https://github.com/secure-systems-lab/securesystemslib/compare/master...in-toto:securesystemslib:master#diff-ae2ca657fc834f3033da3502b45f73a8380f6a6cc2fde77fafb3715fa5237c65R168) to allow independent attributes, and `verify` and `match_keyid` implementations.

The downside is that this makes deserialization harder. You'd have to call either `SSlibKey.from_dict` or `GPGKey.from_dict` based on the format of the passed dict, e.g. in python-tuf's `api.metadata.Root.from_dict`. But I think that's feasible. What do you think?

Reply:
all of those links point to a 1400 LOC diff

Reply:
Hm. They should have pointed to a line in that diff. Let me try again.

Reply:
Updated the links. Now it should work.

Reply:
but I found it: my first questions is, is this just wrapping paper to hide the  fact that GPGSignature and GPGKey are incompatible with the tuf and intoto specs? In that case I'm not fond of the idea of breaking an abstraction that works fine otherwise... 

If you do want to support non-compliant keys, can't the generic Key implementation just hide that ugly detail without forcing clients to change their code?

Reply:
actually, I would not want Key to suddenly use non-compliant metadata in python-tuf... so I guess Key hiding that is out of the question (at least without a "i-want-non-compliant-metadata" flag somewhere).

Further thinking:
* GPGKey (as designed) existing in securesystemslib seems fine if intoto uses it already _but it should not be a Key_ otherwise the meaning of Key is lost
* If you really want a superclass I guess it would be this:
```
class IntotoKey # abstract
class Key(IntotoKey)
class PGPKey(IntotoKey)
```
these are essentially same classes as yours: it's just that I'm calling your SSLibKey Key -- that's the real key class, everything else is wrapping paper for intoto